evalml/pipelines/component_graph.py

```diff
@@ -475,3 +497,31 @@ def __next__(self):
         else:
             self._i = 0
             raise StopIteration
+
+    def _get_parent_y(self, component_name):
+        """Helper for inverse_transform method."""
+        parents = self.get_parents(component_name)
+        return next(iter(p[:-2] for p in parents if '.y' in p), None)
```

Comment:
Yea I think our component graph would break but it would be covered by unit tests at least. I think if a user does something crazy like `.z` or something, an exception about the graph not being connected should be raised. Maybe we can make that clearer in the future.